app/controllers/CategoryGuidanceController.scala

```diff
@@ -38,21 +40,35 @@ class CategoryGuidanceController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   view: CategoryGuidanceView,
   auditService: AuditService,
-  categorisationService: CategorisationService
+  categorisationService: CategorisationService,
+  goodsRecordConnector: GoodsRecordConnector
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
   def onPageLoad(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      request.userAnswers.get(CommodityQuery) match {
-        case Some(_) =>
-          for {
-            _ <- categorisationService.requireCategorisation(request, recordId)
-          } yield Ok(view(recordId))
-        case None    =>
-          Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
-      }
+      (for {
+        ua                    <- categorisationService.requireCategorisation(request, recordId)
+        recordCategorisations <- Future.fromTry(Try(ua.get(RecordCategorisationsQuery).get))
+        categorisationInfo    <- Future.fromTry(Try(recordCategorisations.records.get(recordId).get))
+        scenario               = Scenario.getRedirectScenarios(categorisationInfo)
+        categoryRecord        <- Future.fromTry(Try(CategoryRecord.build(ua, request.eori, recordId).right.get))
+      } yield scenario match {
+        case Category1NoExemptions | StandardNoAssessments =>
+          goodsRecordConnector
+            .updateGoodsRecord(request.eori, recordId, categoryRecord)
+            .map { _ =>
+              Redirect(routes.CategorisationResultController.onPageLoad(recordId, scenario).url)
+            }
+            .recover { case _ =>
```

Comment:
you can put one recover block which encapsulates the whole for comprehension instead of two inside it 

Looks like we only need categoryRecord when it's Category1NoExemptions | StandardNoAssessments so we can do that in there, maybe separate the handling of different cases into a separate function for readability